### ISSUE TITLE
Processing images from video

### ISSUE THREAD

**Issue by bhack** [facebookresearch/vggsfm#9]
Is there any specific setting to take advantage of images from video sequences?

**Comment by jytime** [comment]
Hi @bhack we are preparing for this and should be available in next version.

**Comment by bhack** [comment]
Do you have just another planned version or paper?

**Comment by jytime** [comment]
yes we are planning to release a new version of (a) a hugging face demo that can run on arbitrary input videos and (b) training script. The new version will also reduce the gpu usage by half

**Comment by bhack** [comment]
Do you will have also an option to mask non rigid pixel in video mode?

**Comment by bhack** [comment]
E.g. like the clustering in https://littlepure2333.github.io/GFlow/

**Comment by bhack** [comment]
@jytime So is there something available in the new V2?

**Comment by bhack** [comment]
In the case currently we can autocluster/classify rigid and non rigid points can you at least support an init mask on the dynamic parts in the tracker?

**Comment by jytime** [comment]
Hey,

Currently, we do not have an update on the video processing feature as we are focusing on improving memory efficiency, as the mem problem seems to bother many people.

Regarding non-rigid pixels, you have two options:

(a) Avoid using non-rigid points as query points. You can achieve this by passing a mask to the get_query_points() function https://github.com/facebookresearch/vggsfm/blob/8c47df83eb61071a09b029c784e88431d6ea400e/demo.py#L453

or 

(b)  set all non-rigid points as non-visible, such as done at:

https://github.com/facebookresearch/vggsfm/blob/8c47df83eb61071a09b029c784e88431d6ea400e/demo.py#L236
All the non-visible will be ignored during triangulation.

**Comment by bhack** [comment]
What kind of mask density this is going to require over a sequence? How many query you are preparing?

**Comment by cv-lab-x** [comment]
any schedule to release the version of processing video ?

**Comment by jytime** [comment]
Hi I am doing it now. I am comparing different methods to filter out dynamic objects in the videos. One potential solution is something like @bhack mentioned above. I am also comparing the off-the-shelf motion segmentation methods. Let's see which will win.

**Comment by bhack** [comment]
Yes I've shared other fresh resources and comments in the other issue https://github.com/facebookresearch/vggsfm/issues/24#issuecomment-2226950563

**Comment by jytime** [comment]
Thanks for sharing these all. A lot of distractions recently. Hope I can do my best to share it as soon as possible.

**Comment by jytime** [comment]
Forgot to comment here, please check the section below for this:

https://github.com/facebookresearch/vggsfm/tree/main?tab=readme-ov-file#8-dynamicmoving-objects
